src/stream/src/executor/approx_percentile/global.rs

```diff
@@ -0,0 +1,244 @@
+                        ];
+
+                        if old_row.is_none() {
+                            bucket_state_table.insert(new_row);
+                        } else {
+                            bucket_state_table.update(old_row, new_row);
+                        }
+                    }
+                }
+                Message::Barrier(barrier) => {
+                    // If we haven't received any input, we don't need to update the state.
+                    // This is unless row count state is empty, then we need to persist the state,
+                    // and yield a NULL downstream.
+                    if !received_input && row_count_state.is_some() {
+                        count_state_table.commit(barrier.epoch).await?;
+                        bucket_state_table.commit(barrier.epoch).await?;
+                        yield Message::Barrier(barrier);
+                        continue;
+                    }
+                    // We maintain an invariant, iff row_count_state is none,
+                    // we haven't pushed any data to downstream.
+                    // Naturally, if row_count_state is some,
+                    // we have pushed data to downstream.
+                    let new_output = Self::get_output(
+                        &bucket_state_table,
+                        row_count as u64,
+                        self.quantile,
+                        self.base,
+                    )
+                    .await?;
+                    let percentile_chunk = if row_count_state.is_none() {
+                        StreamChunk::from_rows(
+                            &[(Op::Insert, &[new_output.clone()])],
+                            &[DataType::Float64],
+                        )
+                    } else {
+                        StreamChunk::from_rows(
+                            &[
+                                (Op::UpdateDelete, &[prev_output.clone()]),
+                                (Op::UpdateInsert, &[new_output.clone()]),
+                            ],
+                            &[DataType::Float64],
+                        )
+                    };
+                    prev_output = new_output;
+                    yield Message::Chunk(percentile_chunk);
+
+                    let new_row_count_state = &[Datum::from(ScalarImpl::Int64(row_count))];
+                    if let Some(row_count_state) = row_count_state {
+                        count_state_table.update(row_count_state, new_row_count_state);
+                    } else {
+                        count_state_table.insert(new_row_count_state);
+                    }
+                    row_count_state = Some(new_row_count_state.into_owned_row());
+                    count_state_table.commit(barrier.epoch).await?;
+
+                    bucket_state_table.commit(barrier.epoch).await?;
+
+                    yield Message::Barrier(barrier);
+                }
+                m => yield m,
```

Comment:
Yield the watermark message directly might produce a wrong result. I think we can absorb the watermark in this global executor as well as the local executor

Reply:
Fixed